src/ssh-generator/ssh-proxy.c

```diff
@@ -91,6 +91,100 @@ static int process_unix(const char *path) {
         return 0;
 }
 
+static int skip_ok_port_res(int fd, const char *path, const char *port) {
+        struct timeval oldtv;
+        socklen_t oldtv_size = sizeof(oldtv);
+        if (getsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &oldtv, &oldtv_size) < 0)
+                return log_error_errno(errno, "Failed to get socket receive timeout for %s: %m", path);
+        if (oldtv_size != sizeof(oldtv))
+                return log_error_errno(SYNTHETIC_ERRNO(EIO), "Unexpected size of socket receive timeout for %s: %m", path);
+        struct timeval newtv = {
+                .tv_sec = 5,
+                .tv_usec = 0,
+        };
+        if (setsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &newtv, sizeof(newtv)) < 0)
+                return log_error_errno(errno, "Failed to set socket receive timeout for %s: %m", path);
+
+        /* 10 is the size of "OK 65535\n" */
+        char recv_buf[10];
+        size_t bytes_recv = 0, bytes_avail = 0, pos = 0;
+        static const char expected_prefix[] = "OK ";
+
+        for (;;) {
+                if (pos >= bytes_avail) {
+                        assert(bytes_recv <= bytes_avail);
+                        if (bytes_avail < sizeof(recv_buf)) {
```

Comment:
i'd really suggest inverting this check, i.e. hendle the exceptional case that result in early exit first, so that the expected, common case doesn't have to be indented, and makes it easier for people to follow the codeflow and discern error handling paths from "clean/expected" paths.